numpy/core/_add_newdocs.py

```diff
@@ -6831,6 +6831,11 @@ def add_newdoc_for_scalar_type(obj, fixed_aliases, doc):
     numbers.
     """)
 
+add_newdoc_for_scalar_type('datetime64', [''],
+    """
+    Datetime type.
```

Comment:
This could use a longer explanation. Look at some of the other dtype docstrings in this file.

Reply:
Happy if someone wants to suggest a better description, but I'm struggling to come up with anything myself.

Reply:
"Datetime type, stored as a 64-bit integer. Functionally compatible with `datetime.datetime`"

I think single backticks around `datetime.datetime` will create a link to the python documentation. Should also document timedelta64

Reply:
@dstansby ping

Reply:
Afraid I don't have the bandwidth to pick this up at the moment - if someone else wants to push to my branch or open a replacement PR then feel free.